### ISSUE TITLE
Lenis.scrollTo() - Duration / easing not working properly?

### ISSUE THREAD

**Issue by diegoliv** [darkroomengineering/lenis#22]
First of all, thanks for the best smooth scrolling library out there! Being able to use position: sticky again is liberating.

I'm trying to setup some anchor links smooth scrolling using Lenis.scrollTo() function, but I'm not sure if I understand how `duration` and `easing` works. Here's my current code:

```
  const scrollButtons = document.querySelectorAll('.button-scroll');
  scrollButtons.forEach(button => {
    button.addEventListener('click', e => {
    	e.preventDefault();
    	var target = button.dataset.target;
      lenis.scrollTo(target, {
      	offset: 0, 
        immediate: false, 
        duration: 3,
        easing: (x) => (x < 0.5 ? 8 * x * x * x * x : 1 - pow(-2 * x + 2, 4) / 2)
      });
    });
```

The expectation here is to have the anchor link triggering a scroll with 3 seconds of duration, using a `easeInOutQuart` function for easing. However, the scroll happens extremely fast and the easing function is not applied.

Am I doing something wrong? Can you guys add more details about these properties on the documentation?

**Comment by clementroche** [comment]
Hi thank for your feedback !
Your easing function seems wrong, can you try with `easing: (x) => (x < 0.5 ? 8 * x * x * x * x : 1 - Math.pow(-2 * x + 2, 4) / 2)`

**Comment by diegoliv** [comment]
@clementroche thanks for the answer! I've updated the code with the easing function, but it still looks the same. I'm not seeing any difference.

Apart from the easing function, what about the duration of the scroll? For me that's the biggest issue.

You can see how this looks on my portfolio website here: https://www.diegoliv.works/ (just click on either "about me" or "back to top" button at the footer)

**Comment by arzafran** [comment]
Hello @diegoliv ! thanks for trying our library out! and for coming back with issues for us to perfect it! i was checking the issue and seems like you're using the version 0.1.12 but the easing and duration features were introduced after version 0.2.xx and up, can you please update the library and let me know if it's working correctly?

here's the link to do so:

`https://cdn.skypack.dev/@studio-freight/lenis/bundled`

thanks!

**Comment by diegoliv** [comment]
> Hello @diegoliv ! thanks for trying our library out! and for coming back with issues for us to perfect it! i was checking the issue and seems like you're using the version 0.1.12 but the easing and duration features were introduced after version 0.2.xx and up, can you please update the library and let me know if it's working correctly?
> 
> here's the link to do so:
> 
> `https://cdn.skypack.dev/@studio-freight/lenis/bundled`
> 
> thanks!

Hey @arzafran thanks for the information! I tried to replace the module import source just like you suggested, but now I'm getting an error 404 coming from the script:

![image](https://user-images.githubusercontent.com/5267047/189888751-ea590b7a-fa1e-4317-9c5b-7d9b0d83090a.png)

When trying to access the url that triggered the 404 error directly, I see this:

![image](https://user-images.githubusercontent.com/5267047/189888956-3da24b30-fd58-490f-a35a-d9d325c982d0.png)

Maybe that `/-/` part of the url is triggering the error? I removed it (`https://cdn.skypack.dev/@studio-freight/lenis@v0.2.2-U7EtcputCBPO4ahqkJhA/dist=es2019,mode=raw/bundled`), and then I see this:

![image](https://user-images.githubusercontent.com/5267047/189889303-7da88e4e-6f7c-44a4-be8b-44109952c73d.png)

So yeah, maybe something related to the import url inside the library or something related to Skypack.

Let me know what you find out and thanks for the help!

**Comment by diegoliv** [comment]
Oh, another thing I noticed: on SkyPack, if I check the url `https://www.skypack.dev/view/@studio-freight/locomotive-scroll`, I can see the info about the package, but if I try `https://www.skypack.dev/view/@studio-freight/lenis`, all I see is a "page not found" error.

**Comment by clementroche** [comment]
Can you show how you're calling lenis from skypack ? Are you using an `import` or a `script` tag ? 
This url should work `https://cdn.skypack.dev/@studio-freight/lenis`

**Comment by diegoliv** [comment]
@clementroche I'm using the script tag importing strategy as mentioned on #4:

```
<script type="module">
  import Lenis from "https://cdn.skypack.dev/@studio-freight/lenis@0.1.12";

  const lenis = new Lenis({
    lerp: 0.05,
    smooth: true,
    direction: 'vertical',
  })

  function raf() {
    lenis.raf()
    requestAnimationFrame(raf)
  }
  
  const scrollButtons = document.querySelectorAll('.button-scroll');
  scrollButtons.forEach(button => {
    button.addEventListener('click', e => {
    	e.preventDefault();
    	var target = button.dataset.target;
      lenis.scrollTo(target, {
      	offset: 0, 
        immediate: false, 
        duration: 3,
        easing: (x) => (x < 0.5 ? 8 * x * x * x * x : 1 - pow(-2 * x + 2, 4) / 2)
      });
    });
  });
  
  requestAnimationFrame(raf)
</script>
```

As mentioned, `0.1.12` works great with the script module tag. But `0.2.2` throws the 404 error.

**Comment by clementroche** [comment]
it actually works for me. See [codepen](https://codepen.io/ClementRoche/pen/XWqNRPL)

**Comment by diegoliv** [comment]
> it actually works for me. See [codepen](https://codepen.io/ClementRoche/pen/XWqNRPL)

@clementroche thanks, the code on the Codepen above actually worked! Maybe it was not working before because I had some old code (like the `lerp` property) but was importing the new library?

HOWEVER, now I have another issue with the `lenis.scrollTo()` method. It's not working for all scroll triggers.

So, here's the old code (that works with the `0.1.12` version of Lenis):

```
<script type="module">
  import Lenis from "https://cdn.skypack.dev/@studio-freight/lenis@0.1.12";

  const lenis = new Lenis({
    lerp: 0.05,
    smooth: true,
    direction: 'vertical',
  })

  function raf() {
    lenis.raf()
    requestAnimationFrame(raf)
  }
  
  const scrollButtons = document.querySelectorAll('.button-scroll');
  scrollButtons.forEach(button => {
    button.addEventListener('click', e => {
    	e.preventDefault();
    	var target = button.dataset.target;
      lenis.scrollTo(target, {
      	offset: 0, 
        immediate: false, 
      });
    });
  });
  
  requestAnimationFrame(raf)
</script>
```
This version of the code works fine for both scroll triggers on this page ("About Me" and "Back to Top") - https://www.diegoliv.works/

----

This is the new version of the code, based on your Codepen, that works for me:

```
<script type="module">
  import Lenis from "https://cdn.skypack.dev/@studio-freight/lenis@0.2.2";

  const lenis = new Lenis({
    duration: 1.2,
    easing: (t) => (t === 1 ? 1 : 1 - Math.pow(2, -10 * t)), // https://easings.net
    smooth: true,
    direction: 'vertical',
  })

  function raf(time) {
    lenis.raf(time)
    requestAnimationFrame(raf)
  }

  const scrollButtons = document.querySelectorAll('.button-scroll');
  scrollButtons.forEach(button => {
    button.addEventListener('click', e => {
      e.preventDefault();
      var target = button.dataset.target;
      lenis.scrollTo(target, {
        offset: 0, 
        immediate: false, 
      });
    });
  });

  requestAnimationFrame(raf)
</script>
```
And the scroll behavior works fine on this page: https://diegolivworks.webflow.io/home-lenis. HOWEVER, using the exact same snippet of code to trigger `lenis.scrollTo()` on click of the two triggers on the page ("About Me" and "Back to Top"), I only get the `scrollTo()` function to work for the second trigger (Back to Top). The first trigger (About Me) doesn't trigger `scrollTo()` for some reason.

So, the summary is:
- Lenis `@0.1.12` triggers `scrollTo()` for all scroll triggers on the page
- Lenis `@0.2.2` only triggers `scrollTo()` for the last scroll trigger on the page

I tried picking specifically the first trigger, double checked if the click event was triggering, tried to point directly to the scroll element (`#about`) without using `dataset.target`, tried to change the type of element (from link to just a div) but it's not working for some reason. Any ideas about what could be causing this issue?

**Comment by diegoliv** [comment]
Actually @clementroche I did another test now and the `lenis.scrollTo()` IS ACTUALLY triggering for the first scroll trigger (About me). BUT, instead of scrolling the page to the specified target (footer element with `#about` id), it's just scrolling back to the top.

You can confirm this behavior here - https://diegolivworks.webflow.io/home-lenis. I've added a `console.log()` check to see what is the specified target once you click on the "About Me" button, and it's indeed pointing to the right target.

**Comment by diegoliv** [comment]
@clementroche just tested both snippets of code on another website, and the same behavior happens:

- Lenis `@0.1.12` with `scrollTo()` works perfectly fine here: https://www.alphaminer.xyz/
- Lenis `@0.2.2` with `scrollTo()` is not working here: https://alphaminer-site.webflow.io/home-lenis. All scroll triggers takes you back to the top instead of scrolling to the right positions.

**Comment by BleXor** [comment]
Had/have the same issue (v ^0.2.1) with `scrollTo()` with element/selector as parameter - it scrolls to top of page .. "Fixed" it by using `element.offsetTop` as the parameter instead. See:

```
const targetEl = document.getElementById(targetId);
targetEl && this.lenis.scrollTo(targetEl.offsetTop, { ... });
```

**Comment by diegoliv** [comment]
@BleXor THANK YOU! Yes, `targetEl.scrollTop` works perfectly. Glad to know this is not only happening to me. Just replaced my code with your suggestion on my personal portfolio and now everything looks great.

@clementroche you can still see the issue on the second website I included on my last comment above for a sample of the issue if you need to investigate this further.

**Comment by clementroche** [comment]
Hi you were right, it's fixed in v0.2.4👌
See [commit](https://github.com/studio-freight/lenis/commit/6d1c1f87996f7764c0569aa13c49b11bc4f3aa01).

**Comment by diegoliv** [comment]
@clementroche thanks, just updated the second website and it works perfectly now! 🥇
